main/acle.md

```diff
@@ -1854,6 +1856,142 @@ or compiler-generated functions.
 
 This attribute does not apply to AArch64.
 
+## Function Multi Versioning
+
+The specification for Function Multi Versioning is in `BETA` state and may
+change or be extended in the future.
+
+Function Multi Versioning provides a convenient way to select the most
+appropriate version of a function at runtime. All versions of the function may
+be in the final binary. The compiler generates all supported versions and the
+runtime makes the selection at load time.
+
+The following attributes trigger the multi variant code generation:
+`target_version`, `target_clones`. These attributes can't be mixed. The
+"default" variant means the version of the function that would be generated
+without these attributes.
+
+This feature may be turned off in compile time. In this case the "default"
+variant shall be used.
+
+`__ARM_FEATURE_FUNCTION_MULTI_VERSIONING` is defined to 1 if available.
+
+``` c
+    __attribute__((target_version("name")))
+```
+
+when applied to a function it will become one of the versions. Function with the
+same name may exist with multiple versions in the same translation unit. One
+"default" version of the  function is required to be provided. Implicitly, when
+without attribute, or explicitly providing the "default" in the attribute. If
+only the "default" variant exist it should be linked directly.
+
+``` c 
+    __attribute__((target_clones("name",...)))
+```
+
+when applied to a function the compiler emits multiple versions based on the
+arguments. One of them is implicitly the "default". If a name is not recognized
+the compiler should ignore it. If the default matches with an other explicitly
+provided version the compiler can emit only one function instead of the two.
+
+Dependent features names could be joined by the `+` sign.
+
+None of these attributes will enable the corresponding ACLE features.
+
+These attributes have no effect on the calling convention, all versions must
+use the same calling convertion.
```

Comment:
Given that these sentences are additional rules, and that the top of the paragraph have also rules, I would organize this paragraph as follows:

1. Introduce the concept of function multiversioning
2. introduce the names of the 2 attributes
3. introduce the rules that are common to both attributes, with enumerated list
4. introduce the rules that are specific to each attributes, with enumeration lists that are starting from the previous set of enumeration list. By doing so, you have unique numbers to refer to in the paragraph

Reply:
ack